FT003-feat-stworzenie-menu-bocznego

damiankubecki opened the pull request:
podgląd: https://ft003-feat-stworzenie-menu-bocznego--meetek-envelo.netlify.app/

damiankubecki committed:
fix: merge

damiankubecki committed:
feat: updated home component

damiankubecki committed:
feat: updated home module

damiankubecki committed:
feat: updated imports

damiankubecki committed:
fix: fixed styles and updated inital menu state

piodaw commented on src/app/core/theme.component.ts:
```diff
@@ -24,6 +28,8 @@ import { ToggleComponent } from '@shared/ui/toggle/toggle.component';
     MatButtonModule,
     HeaderComponent,
+    MenuComponent,
+    CommonModule,
```
zamiast CommonModule importujmy wszystko osobno

piodaw commented on src/app/core/theme.component.ts:
```diff
@@ -99,6 +114,15 @@ export default class ThemeComponent {
+  menuService = inject(MenuService);
+  isMenuActive = this.menuService.menu$.pipe(map(menu => menu.isActive));
+  userMockMenu: MenuInputs = {
+    userData: { fullName: 'test user hahahadsdssdsdsdsddh', email: 'test@gmail.com', initials: 'TU' },
```
te dane są potrzebne?

damiankubecki replied:
pokazuje tutaj przykładowe użycie menu, bez danych nie da się go wyświetlić

piodaw commented on src/app/core/theme.component.ts:
```diff
@@ -99,6 +114,15 @@ export default class ThemeComponent {
+  };
+
   onToggleChange(isChecked: boolean) {
     console.log(isChecked);
```
console.loga nie powinno być

damiankubecki replied:
to jest `theme.component`, więc console.logi wrzucamy żeby pokazać działanie + ten console.log pochodzi w ogole z innego taska

piodaw commented on src/app/features/home/home.component.scss:
```diff
@@ -0,0 +1,42 @@
+
+    .menu-container {
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      left: 0;
```
proponuję użyć np: top, left i transform

damiankubecki replied:
menu musi się wyświetlać od samej góry do samego dołu ekranu

piodaw commented on src/app/shared/ui/menu/menu.component.html:
```diff
@@ -0,0 +1,54 @@
+      <p>Dark mode</p>
+    </div>
+    <button (click)="closeMenu()" class="close-button">
+      <mat-icon aria-hidden="false" fontIcon="close" [style]="{ padding: '10px' }"></mat-icon>
```
czy to aria-hidden jest konieczne?

damiankubecki replied:
poprawiam

piodaw commented on src/app/shared/ui/menu/menu.component.html:
```diff
@@ -0,0 +1,54 @@
+          <mat-expansion-panel-header>
+            <mat-panel-title>
+              <p class="item-content">
+                <mat-icon aria-hidden="false" [fontIcon]="item.icon"></mat-icon>
```
jw

piodaw commented on src/app/shared/ui/menu/menu.component.scss:
```diff
@@ -0,0 +1,213 @@
+  }
+}
+
+@media (min-width: '567px') {
```
Korzystaj ze stałych

damiankubecki replied:
nasze breakpointy nie są `mobile first` 🧐 do przegadania z teamem

damiankubecki replied:
cssy przerobione, użyte stałe z `breakpoints.scss`

piodaw commented on src/styles.scss:
```diff
@@ -126,3 +131,12 @@ body {
+  color: var(--surface);
+}
+
+@media (min-width: '578px') {
+  mat-expansion-panel.menu .mat-expansion-indicator::after {
+    color: var(--text);
+  }
+ }
```
korzystaj ze stałych

damiankubecki committed:
fix: updated theme component and menu component

This commit removes CommonModule from the theme component and aria-hidden tags in menu component

EMEZYK commented on src/app/shared/ui/menu/menu-items.ts:
```diff
@@ -0,0 +1,50 @@
+import { HOME_PATHS } from 'src/app/features/home/home-paths';
+
+interface MenuItem {
+  name: string;
+  icon: string;
+  path?: string;
+  subpages?: { name: string; path: string }[];
+  onlyMobile?: boolean;
+}
+
+export interface MenuItems {
```
proponuję wynieść interfejsy do oddzielnego pliku

damiankubecki replied:
wyniosę

EMEZYK commented on src/app/shared/ui/menu/menu.component.html:
```diff
@@ -0,0 +1,54 @@
+      <p>Dark mode</p>
+    </div>
+    <button (click)="closeMenu()" class="close-button">
+      <mat-icon fontIcon="close" [style]="{ padding: '10px' }"></mat-icon>
```
sugeruję wynieść to stylowanie do pliku scss

EMEZYK commented on src/app/shared/ui/menu/menu.component.html:
```diff
@@ -0,0 +1,54 @@
+  <div *ngIf="user.userData as userData" class="user-info-container">
+    <app-user-initials-button [initials]="userData.initials"></app-user-initials-button>
+    <div>
+      <p class="title">{{ userData.fullName }}</p>
+      <p class="subtitle">{{ userData.email }}</p>
```
nazwy klas są trochę mało intuicyjne, zmieniłabym na 'name' / 'fullName'  i 'email'

EMEZYK commented on src/app/shared/ui/menu/menu.component.html:
```diff
@@ -0,0 +1,54 @@
+<nav class="wrapper">
+  <div class="top-section">
+    <div class="toggle-mode-container">
+      <mat-slide-toggle></mat-slide-toggle>
+      <p>Dark mode</p>
```
ten toggle jest używany również przy formularzu logowania; może wrzućmy do backlogu task, żeby wydzielić ten toggle do odrębnego komponentu i zmienić wtedy w tych 2 miejscach?  i wtedy tutaj dodac jeszcze komentarz z nr nowego taska

damiankubecki replied:
racja

EMEZYK commented on src/app/shared/ui/menu/menu.component.scss:
```diff
@@ -0,0 +1,213 @@
+      color: var(--surface);
+      white-space: nowrap;
+      text-overflow: ellipsis;
+      white-space: nowrap;
```
duplikat (patrz 2 linijki wyżej)

damiankubecki replied:
czujne oko 😎

EMEZYK commented on src/app/shared/ui/menu/menu.component.html:
```diff
@@ -0,0 +1,54 @@
+          </mat-expansion-panel-header>
+          <mat-list-item
+            *ngFor="let subpage of item.subpages"
+            class="small"
```
ta nazwa klasy też jest trochę mało intuicyjna dla mnie

EMEZYK commented on src/app/shared/ui/menu/menu.component.scss:
```diff
@@ -0,0 +1,213 @@
+      background-color: transparent;
+      box-shadow: none;
+
+      &.mat-expansion-panel:not(.mat-expanded):not([aria-disabled='true']) .mat-expansion-panel-header:hover {
```
niezły ten zapis :D

EMEZYK commented on src/app/core/store/user.state.ts:
```diff
@@ -1,11 +1,11 @@
-      role: 'user'
-    }
+  user: {
+    id: NaN,
```
NaN? :)

damiankubecki replied:
Wszystko podświetlone na zielono bo ktoś zapomniał tutaj użyć pretteria ;( Ode mnie żadnych zmian tu nie ma

EMEZYK commented on src/app/shared/ui/menu/menu.component.html:
```diff
@@ -0,0 +1,54 @@
+      </ng-container>
+
+      <ng-template #withoutSubpages>
+        <mat-list-item *ngIf="!item.subpages?.length" [class.active]="item.path === currentRoute">
```
a dlaczego tutaj stosujesz "!" ?

damiankubecki replied:
tego nie powinno byc, usuwam

EMEZYK commented on src/app/shared/ui/menu/menu.component.ts:
```diff
@@ -0,0 +1,71 @@
+import { UserInitialsButtonComponent } from '../user-initials-button/user-initials-button.component';
+import { UserMenuComponent, UserMenuInputs } from '../user-menu';
+
+export interface MenuInputs {
```
jw. wynieśmy do jednego oddzielnego pliku menu,interface.ts  te interfejsy, żeby nie mieszać tych funkcjonalności w jednym pliku

EMEZYK commented on src/app/shared/ui/menu/menu.component.ts:
```diff
@@ -0,0 +1,71 @@
+
+export interface MenuInputs {
+  role: UserRole;
+  userData: Required<UserMenuInputs>;
```
O, to required ciekawe, jak to działa?

damiankubecki replied:
Sprawia, że wszystkie pola interfejsu stają się `required` ;) ma to tutaj zastosowanie  bo interface `UserMenuInputs` wykorzystwany jest w innym miejscu i wszystkie pola są zapisane jako `optional`. 
Jakbyśmy chcieli zrobić to samo, ale w drugą stronę wtedy można użyć `Partial<>` 🙂

EMEZYK commented on src/app/shared/ui/menu/menu.component.ts:
```diff
@@ -0,0 +1,71 @@
+    RouterModule,
+    UserInitialsButtonComponent,
+    UserMenuComponent,
+    AsyncPipe,
```
chyba nigdzie nie używasz async pipe? można wyrzucić

EMEZYK commented on src/app/shared/ui/menu/menu.component.ts:
```diff
@@ -0,0 +1,71 @@
+        filter(event => event instanceof NavigationEnd),
+        map(() => this.router.url)
+      )
+      .subscribe(route => {
```
a co z odsubskrybowaniem się?

damiankubecki replied:
zapomnialem 😑 poczekajmy na zmergowanie taska, w którym dodałem `use-destroy.hook` to i tu się wykorzysta😉

EMEZYK commented on src/app/shared/ui/menu/menu.component.ts:
```diff
@@ -0,0 +1,71 @@
+import { MenuService } from './menu.service';
+import { menuItems } from './menu-items';
+import { UserRole } from '@core/store/user.interfaces';
+import { UserInitialsButtonComponent } from '../user-initials-button/user-initials-button.component';
```
wrzucmy to do pliku index.ts, zeby skrocic ten import

EMEZYK commented on src/app/shared/ui/menu/menu.component.html:
```diff
@@ -0,0 +1,54 @@
+    </button>
+  </div>
+
+  <div *ngIf="user.userData as userData" class="user-info-container">
```
moze dodajmy tutaj user?.userData , nizej w role podobnie

damiankubecki replied:
nie bardzo rozumiem. możesz rozwinąć myśl? 🤔

EMEZYK replied:
sprawdzilam, ze jeśli user byłby nullem, to wywali nam ponizszy błąd;  a dodając '?' możemy się przed tym zabezpieczyć

![image](https://user-images.githubusercontent.com/87782933/217016916-13f44724-a02f-4306-9a5b-8cb349d5b51e.png)

damiankubecki replied:
oki, poprawione

damiankubecki committed:
fix: updated styles

damiankubecki committed:
fix: updated menu

damiankubecki committed:
fix: updated menu template

damiankubecki committed:
fix: updated menu styles

Commit:
fix: updated menu component

Commit:
fix: merge

Commit:
fix: removed unnecessary imports
